src/sa1hybridizer/index.js

```diff
@@ -0,0 +1,396 @@
+%define_sprite_table("1632", $1632, $75A4)
+%define_sprite_table("163E", $163E, $33FA)
+%define_sprite_table("164A", $164A, $75BA)
+%define_sprite_table("1656", $1656, $75D0)
+%define_sprite_table("1662", $1662, $75EA)
+%define_sprite_table("166E", $166E, $7600)
+%define_sprite_table("167A", $167A, $7616)
+%define_sprite_table("1686", $1686, $762C)
+%define_sprite_table("186C", $186C, $7642)
+%define_sprite_table("187B", $187B, $3410)
+%define_sprite_table("190F", $190F, $7658)
+
+%define_sprite_table("1938", $7FAF00, $418A00)
+%define_sprite_table("7FAF00", $7FAF00, $418A00)
+
+%define_sprite_table("1FD6", $1FD6, $766E)
+%define_sprite_table("1FE2", $1FE2, $7FD6)
+`;
+
+const bwram_defines = `
+macro define_bwram(addr, bwram)
+    if read1($00FFD5) == $23
+        !<addr> = $<bwram>
+    else
+        !<addr> = $<addr>
+    endif
+endmacro
+%define_bwram(7F9A7B, 418800) ; ends at 7F9C7A
+%define_bwram(700800, 41A000) ; ends at 7027FF
+if read1($00FFD5) == $23
+    !map16_lo_by = $400000
+    !map16_hi_by = $410000
+    !save_mem = $41C000
+else
+    !map16_lo_by = $7E0000
+    !map16_hi_by = $7F0000
+    !save_mem = $700000
+endif"
+`;
+
+const sprite_addr_list = [0x7fab10, 0x7fab1c, 0x7fab28, 0x7fab34, 0x7fab9e, 0x7fab40, 0x7fab4c, 0x7fab58, 0x7fab64, 0x7fac00,
+    0x7fac08, 0x7fac10, 0x9e, 0xaa, 0xb6, 0xc2, 0xd8, 0xe4, 0x14c8, 0x14d4, 0x14e0, 0x14ec, 0x14f8,
+    0x1504, 0x1510, 0x151c, 0x1528, 0x1534, 0x1540, 0x154c, 0x1558, 0x1564, 0x1570, 0x157c, 0x1588,
+    0x1594, 0x15a0, 0x15ac, 0x15b8, 0x15c4, 0x15d0, 0x15dc, 0x15ea, 0x15f6, 0x1602, 0x160e, 0x161a,
+    0x1626, 0x1632, 0x163e, 0x164a, 0x1656, 0x1662, 0x166e, 0x167a, 0x1686, 0x186c, 0x187b, 0x190f,
+    0x1938, 0x7faf00, 0x1fd6, 0x1fe2];
+
+const WordType = {
+    OTHER: -1,
+    ADDR: 1,
+    COMMA: 2
+}
+
+class WordTuple {
+    constructor(type, word, index) {
+        this.type = type;
+        this.word = word;
+        this.index = index;
+    }
+}
+
+Object.freeze(WordType);
+
+function normalizeHexStringToHexString(str) {
+    // this seems rather useless (and it may be) but it ensures consitency
+    return parseInt(str, 16).toString(16).toUpperCase();
+}
+
+function check_bwram(word) {
+    let bwram_word = parseInt(word, 16)
+    let bwram_remapped_list = [0x7F9A7B, 0x7027FF]          // Wiggler's segment buffer, Expansion area planned for SMW hacks
+    let map16_lo_by = [0x7EC800, 0x7EFFFF]                // Map16 low byte plus Overworld related data.
+    let map16_hi_by = [0x7FC800, 0x7FFFFF]                // Map16 high byte.
+    let save_mem = [0x700000, 0x7007FF]                // Original save memory (2 kB big). Not everything is used
+    let bwram_list = [map16_lo_by, map16_hi_by, save_mem]
+    let bwram_indexes = bwram_list.map(x => {
+        let [b, e] = x;
+        return b <= bwram_word <= e;
+    });
+    let subs = ['map16_lo_by', 'map16_hi_by', 'save_mem']
+    if (bwram_indexes.some(x => x)) {
+        let true_index = bwram_indexes.indexOf(True)
```

Comment:
`True` is undefined, `true` is a literal boolean.
```suggestion
        let true_index = bwram_indexes.indexOf(true)
```

Reply:
Fixed in 010b017cb417f4e3d9f01b938abf4a2a9697f87c